x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -305,21 +307,23 @@ public void clear() {
         clearCircuitBreaker();
     }
 
-    private long ramBytesUsedInFlight() {
+    protected long ramBytesUsedInFlight() {
         return RamUsageEstimator.sizeOf(keyToSequences) + RamUsageEstimator.sizeOf(stageToKeys);
     }
 
-    private long ramBytesUsedCompleted() {
+    protected long ramBytesUsedCompleted() {
         return RamUsageEstimator.sizeOfCollection(completed);
     }
 
     private void addMemory(long bytes, String label) {
-        totalRamBytesUsed += bytes;
```

Comment:
In case of breaker failure, the memory is not added to the circuit breaker, so the addition to `totalRamBytesUsed` should be done only after `addEstimateBytesAndMaybeBreak` succeeds.

Reply:
This is a good point, should this be relatively easy reproducible?